internal/engine/dockerprune/docker_pruner.go

```diff
@@ -203,25 +202,85 @@ func (dp *DockerPruner) deleteOldImages(ctx context.Context, maxAge time.Duratio
 					inspect.ID, strings.Join(inspect.RepoTags, ", "))
 				continue
 			}
-			toDelete[inspect.ID] = uint64(inspect.Size)
+			result = append(result, inspect)
+		}
+	}
+	return result
+}
+
+// Return all image objects that aren't in the N
+// most recently used for each tag.
+func (dp *DockerPruner) keepMostRecentInspects(ctx context.Context, inspects []types.ImageInspect, keepRecent int, selectors []container.RefSelector) []types.ImageInspect {
```

Comment:
This function signature is confusing because the inspects you're "keeping" are actually the ones you're NOT returning. Maybe frame this is as `FilterOutNMostRecentInspects`?

Reply:
done